src/client/service/identitiesInfoGet.ts

```diff
@@ -18,40 +22,90 @@ function identitiesInfoGet({
   authenticate: Authenticate;
 }) {
   return async (
-    call: grpc.ServerUnaryCall<identitiesPB.Provider, identitiesPB.Provider>,
-    callback: grpc.sendUnaryData<identitiesPB.Provider>,
+    call: grpc.ServerWritableStream<
+      identitiesPB.ProviderSearch,
+      identitiesPB.Info
+    >,
   ): Promise<void> => {
+    const genWritable = grpcUtils.generatorWritable(call);
     try {
-      const response = new identitiesPB.Provider();
       const metadata = await authenticate(call.metadata);
       call.sendMetadata(metadata);
       const {
-        providerId,
+        providerIds,
+        identityId,
       }: {
-        providerId: ProviderId;
+        providerIds: Array<ProviderId>;
+        identityId: IdentityId;
       } = validateSync(
         (keyPath, value) => {
           return matchSync(keyPath)(
-            [['providerId'], () => validationUtils.parseProviderId(value)],
+            [['providerIds'], () => value.map(validationUtils.parseProviderId)],
+            [['identityId'], () => validationUtils.parseIdentityId(value)],
             () => value,
           );
         },
         {
-          providerId: call.request.getProviderId(),
+          providerIds: call.request.getProviderIdList(),
+          identityId: call.request.getIdentityId(),
         },
       );
-      const provider = identitiesManager.getProvider(providerId);
-      if (provider !== undefined) {
-        const identities = await provider.getAuthIdentityIds();
-        response.setProviderId(providerId);
-        if (identities.length !== 0) {
-          response.setIdentityId(identities[0]);
+      // Process options that were set
+      if (providerIds.length === 0) {
+        Object.keys(identitiesManager.getProviders()).forEach((id) =>
+          providerIds.push(id as ProviderId),
+        );
+      }
+      const searchTerms = call.request.getSearchTermList();
+      const getDisconnected = call.request.getDisconnected();
+      if (getDisconnected) {
+        // Currently this command performs the same way regardless of whether
+        // this option is set (i.e. always disconnected)
+      }
+      const identities: Array<IdentityData | undefined> = [];
+      for (const providerId of providerIds) {
+        // Get provider from id
+        const provider = identitiesManager.getProvider(providerId);
+        if (provider === undefined)
+          throw new identitiesErrors.ErrorProviderMissing();
```

Comment:
@CMCDragonkai can potentially confirm this, but based on comments on other PRs, I think we're preferring the style of:

```ts
if (provider === undefined) {
  throw new identitiesErrors.ErrorProviderMissing();
}
```

Single line conditionals are fine too:

```ts
if (provider === undefined) throw new identitiesErrors.ErrorProviderMissing();
```

but I believe the linter will cause a newline because this line would be too long.

Reply:
Yeah I think that was the linter - I'll add in brackets